libres/lib/enkf/enkf_main.cpp

```diff
@@ -1055,25 +1055,109 @@ static void enkf_main_analysis_update_with_rowscaling(
+      nodes which are updated will be fetched from the new target
+      case, and nodes which are not updated will be manually copied
+      over there.
+    */
 
-   The reason for going via these temporary structures is to support
-   deactivating observations which should not be used in the update
-   process.
-  */
+    if (target_fs != source_fs) {
+        stringlist_type *param_keys =
+            ensemble_config_alloc_keylist_from_var_type(ensemble_config,
+                                                        PARAMETER);
+        for (int i = 0; i < stringlist_get_size(param_keys); i++) {
+            const char *key = stringlist_iget(param_keys, i);
+            enkf_config_node_type *config_node =
+                ensemble_config_get_node(ensemble_config, key);
+            enkf_node_type *data_node = enkf_node_alloc(config_node);
+            for (int j = 0; j < int_vector_size(ens_active_list); j++) {
+                node_id_type node_id;
+                node_id.iens = int_vector_iget(ens_active_list, j);
+                node_id.report_step = 0;
+
+                enkf_node_load(data_node, source_fs, node_id);
+                enkf_node_store(data_node, target_fs, node_id);
+            }
+            enkf_node_free(data_node);
+        }
+        stringlist_free(param_keys);
+    }
+}
+
+static bool assert_update_viable(const analysis_config_type *analysis_config,
+                                 enkf_fs_type *source_fs, int total_ens_size,
+                                 const local_updatestep_type *updatestep) {
+    state_map_type *source_state_map = enkf_fs_get_state_map(source_fs);
+    const int active_ens_size =
+        state_map_count_matching(source_state_map, STATE_HAS_DATA);
+
+    if (!analysis_config_have_enough_realisations(
+            analysis_config, active_ens_size, total_ens_size)) {
+        fprintf(stderr,
+                "** ERROR ** There are %d active realisations left, which is "
+                "less than the minimum specified - stopping assimilation.\n",
+                active_ens_size);
+        return false;
+    }
+
+    // exit if multi step update
+    if ((local_updatestep_get_num_ministep(updatestep) > 1) &&
+        (analysis_config_get_module_option(analysis_config,
+                                           ANALYSIS_ITERABLE))) {
+        util_exit("** ERROR: Can not combine iterable modules with multi step "
+                  "updates - sorry\n");
+    }
+    return true;
+}
+
+bool enkf_main_smoother_update(enkf_main_type *enkf_main,
+                               enkf_fs_type *source_fs,
+                               enkf_fs_type *target_fs) {
+    time_map_type *time_map = enkf_fs_get_time_map(source_fs);
+
+    int last_step = time_map_get_last_step(time_map);
+    if (last_step < 0)
+        last_step = model_config_get_last_history_restart(
+            enkf_main_get_model_config(enkf_main));
+
+    int_vector_type *step_list = int_vector_alloc(0, 0);
```

Comment:
I see issue #2437 is working on chinging ```T_vector_type``` to ```std::vector<T>```. meaning this will need to be replaced shortly

Reply:
I see it already got approved just need to pass styling tests, so suggest you wait for merger